app/uk/gov/hmrc/apiplatformevents/scheduled/SendEventNotificationsJob.scala

```diff
@@ -0,0 +1,91 @@
+import play.modules.reactivemongo.ReactiveMongoComponent
+import uk.gov.hmrc.apiplatformevents.connectors.{EmailConnector, ThirdPartyApplicationConnector}
+import uk.gov.hmrc.apiplatformevents.models.NotificationStatus.{FAILED, SENT}
+import uk.gov.hmrc.apiplatformevents.models.ReactiveMongoFormatters.PpnsCallBackUriUpdatedEventFormats
+import uk.gov.hmrc.apiplatformevents.models.common.EventType.PPNS_CALLBACK_URI_UPDATED
+import uk.gov.hmrc.apiplatformevents.models.{Notification, PpnsCallBackUriUpdatedEvent}
+import uk.gov.hmrc.apiplatformevents.repository.{ApplicationEventsRepository, NotificationsRepository}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.lock.{LockKeeper, LockRepository}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+@Singleton
+class SendEventNotificationsJob @Inject()(override val lockKeeper: SendEventNotificationsJobLockKeeper,
+                                          jobConfig: SendEventNotificationsJobConfig,
+                                          applicationEventsRepository: ApplicationEventsRepository,
+                                          notificationsRepository: NotificationsRepository,
+                                          emailConnector: EmailConnector,
+                                          thirdPartyApplicationConnector: ThirdPartyApplicationConnector)
+                                         (implicit mat: Materializer) extends ScheduledMongoJob {
+
+  override def name: String = "SendEventNotificationsJob"
+  override def interval: FiniteDuration = jobConfig.interval
+  override def initialDelay: FiniteDuration = jobConfig.initialDelay
+  override val isEnabled: Boolean = jobConfig.enabled
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  override def runJob(implicit ec: ExecutionContext): Future[RunningOfJobSuccessful] = {
+    applicationEventsRepository
+      .fetchEventsToNotify[PpnsCallBackUriUpdatedEvent](PPNS_CALLBACK_URI_UPDATED)
+      .runWith(Sink.foreachAsync[PpnsCallBackUriUpdatedEvent](jobConfig.parallelism)(sendEventNotification))
+      .map(_ => RunningOfJobSuccessful)
+      .recoverWith {
+        case NonFatal(e) =>
+          Future.failed(RunningOfJobFailed(name, e))
+      }
+  }
+
+  private def sendEventNotification(event: PpnsCallBackUriUpdatedEvent)(implicit ec: ExecutionContext): Future[Unit] = {
+    (for {
+      app <- thirdPartyApplicationConnector.getApplication(event.applicationId)
+      _ <- emailConnector.sendPpnsCallbackUrlChangedNotification(app.name, event.newCallbackUrl, event.eventDateTime, app.adminEmails)
+      _ <- notificationsRepository.createEntity(Notification(event.id, now(UTC), SENT))
+    } yield ()) recoverWith {
+      case NonFatal(e) =>
+        Logger.error(s"Failed to send email notification for event ID ${event.id}", e)
+        notificationsRepository.createEntity(Notification(event.id, now(UTC), FAILED)).map(_ => ())
```

Comment:
Currently it doesn't support retries. So it's recorded as either SENT (above), or FAILED. And next time it won't be processed because it's already in the notifications collection.